hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/footer/DeleteActionsLayout.java

```diff
@@ -258,46 +257,41 @@ private void deleteTargetTag(final Component source) {
 
         final DistributionSetIdName dsInBulkUpload = managementUIState.getTargetTableFilters().getBulkUpload()
                 .getDsNameAndVersion();
         if (isDsInUseInBulkUpload(distributionIdNameSet, dsInBulkUpload)) {
             distributionIdNameSet.remove(dsInBulkUpload);
         }
 
-        if (!distributionIdNameSet.isEmpty()) {
-
+        if (distributionIdNameSet.isEmpty()) {
+            return;
+        }
+        /*
+         * Flags to identify whether all dropped distributions are already in
+         * the deleted list (or) some distributions are already in the deleted
+         * distribution list.
+         */
+        final int existingDeletedDistributionsSize = managementUIState.getDeletedDistributionList().size();
+        managementUIState.getDeletedDistributionList().addAll(distributionIdNameSet);
+        final int newDeletedDistributionsSize = managementUIState.getDeletedDistributionList().size();
+        if (newDeletedDistributionsSize == existingDeletedDistributionsSize) {
             /*
-             * Flags to identify whether all dropped distributions are already
-             * in the deleted list (or) some distributions are already in the
-             * deleted distribution list.
+             * No new distributions are added, all distributions dropped now are
+             * already available in the delete list. Hence display warning
+             * message accordingly.
              */
-            final int existingDeletedDistributionsSize = managementUIState.getDeletedDistributionList().size();
-            managementUIState.getDeletedDistributionList().addAll(distributionIdNameSet);
-            final int newDeletedDistributionsSize = managementUIState.getDeletedDistributionList().size();
-            if (newDeletedDistributionsSize == existingDeletedDistributionsSize) {
-                /*
-                 * No new distributions are added, all distributions dropped now
-                 * are already available in the delete list. Hence display
-                 * warning message accordingly.
-                 */
-                notification.displayValidationError(i18n.get("message.targets.already.deleted"));
-            } else if (newDeletedDistributionsSize - existingDeletedDistributionsSize != distributionIdNameSet.size()) {
-                /*
-                 * Not the all distributions dropped now are added to the delete
-                 * list. There are some distributions are already there in the
-                 * delete list. Hence display warning message accordingly.
-                 */
-                notification.displayValidationError(i18n.get("message.dist.deleted.pending"));
-            }
+            notification.displayValidationError(i18n.get("message.targets.already.deleted"));
+        } else if (newDeletedDistributionsSize - existingDeletedDistributionsSize != distributionIdNameSet.size()) {
```

Comment:
Extract conditions to new private method. So you can express the comment with a proper method name